src/boot/efi/boot.c

```diff
@@ -138,24 +137,20 @@ static void cursor_right(
 }
 
 static BOOLEAN line_edit(
-                const CHAR16 *line_in,
-                CHAR16 **line_out,
+                CHAR16 **line_in,
                 UINTN x_max,
                 UINTN y_pos) {
 
         _cleanup_freepool_ CHAR16 *line = NULL, *print = NULL;
         UINTN size, len, first = 0, cursor = 0, clear = 0;
 
-        assert(line_out);
+        assert(line_in);
 
-        if (!line_in)
-                line_in = L"";
-
-        len = StrLen(line_in);
+        len = *line_in ? StrLen(*line_in) : 0;
         size = len + 1024;
         line = xnew(CHAR16, size);
         print = xnew(CHAR16, x_max + 1);
-        StrCpy(line, line_in);
+        StrCpy(line, *line_in ?: L"");
```

Comment:
we probably should move strempty() into the fundamental stuff shared between userspace and efispace